src/tiledb/cloud/sql/tiledb_cursor.py

```diff
@@ -0,0 +1,117 @@
+import pyarrow as pa
+
+from tiledb.cloud.sql._execution import exec
+from tiledb.cloud.sql.db_api_exceptions import DataError
+from tiledb.cloud.sql.db_api_exceptions import NotSupportedError
+from tiledb.cloud.sql.db_api_exceptions import ProgrammingError
+
+
+def _getDBType(dtype):
+    if pa.types.is_timestamp(dtype):
+        return "DATETIME"
+    elif pa.types.is_boolean(dtype):
+        return "BOOLEAN"
+    elif pa.types.is_string(dtype):
+        return "STRING"
+    elif pa.types.is_integer(dtype) or pa.types.is_floating(dtype):
+        return "NUMBER"
+    return "UNKNOWN"
+
+
+class Cursor:
+    def __init__(self):
+        self._results = None
+        self._row_index = 0
+        self.arraysize = 1
+        self._description_cache = None
+
+    def executemany(self, query):
+        self.execute(query)
+
+    def execute(self, query):
+        try:
+            self._results = exec(query=query, raw_results=True)
+            self._description_cache = None
+        except Exception as e:
+            self._results = None
+            raise DataError(f"Error executing query: {e}") from e
+
+    def fetchmany(self, size=-1):
+        if self._results is None or len(self._results) == 0:
+            return None
+
+        if size == -1:
+            size = self.arraysize
+
+        if size + self._row_index > len(self._results):
+            # give all that is remaining
+            size = len(self._results) - self._row_index
+
+        if size == 0:
+            # There are no more records
+            return []
+
+        rows = self._results[self._row_index : self._row_index + size]
+        self._row_index += size
+        return rows.to_pylist()
+
+    def setinputsizes(self, sizes):
+        raise NotSupportedError("Operation not supported")
+
+    def setoutputsize(self, size, column):
+        raise NotSupportedError("Operation not supported")
+
+    def fetchone(self):
+        result = self.fetchmany(1)
+        if len(result) == 0:
+            return None
+        return result[0]
+
+    def description(self):
+        if self._description_cache is not None:
+            return self._description_cache
+
+        if self._results is None or len(self._results) == 0:
+            return None
+
+        description = [
+            (field.name, _getDBType(field.type)) for field in self._results.schema
+        ]
+        self._description_cache = description
+        return description
+
+    def rowcount(self):
+        if self._results is None or len(self._results) == 0:
```

Comment:
I think this should be 0 when there are specifically 0 results (as opposed to the `_results is None` case, which should still be −1).